asyncpg/connection.py

```diff
@@ -1476,7 +1476,11 @@ def _get_reset_query(self):
             _reset_query.append('SELECT pg_advisory_unlock_all();')
         if caps.sql_close_all:
             _reset_query.append('CLOSE ALL;')
-        if caps.notifications and caps.plpgsql:
+        if (
+            caps.notifications and
+            caps.plpgsql and
+            len(self._listeners.keys()) != 0
```

Comment:
This doesn't account for any potential listeners created with `LISTEN` instead of with the connection API.